src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -152,6 +152,33 @@ export const pinTokensWithBalanceAtTop = ({
 export const sumTokensUiUsdBalance = (tokens: TokenUi[]): number =>
 	tokens.reduce((acc, token) => acc + (token.usdBalance ?? 0), 0);
 
+/**
+ * Calculates total USD balance of mainnet tokens from the provided tokens list.
+ *
+ * @param $tokens - The list of tokens for filtering by network env and total USD balance calculation.
+ * @param $balancesStore - The balances data for the tokens.
+ * @param $exchanges - The exchange rates data for the tokens.
+ * @returns The sum of mainnet tokens USD balance.
+ *
+ */
+export const sumMainnetTokensUsdBalance = ({
+	$tokens,
+	$balances,
+	$exchanges
+}: {
+	$tokens: Token[];
+	$balances: CertifiedStoreData<BalancesData>;
+	$exchanges: ExchangesData;
+}) =>
+	$tokens.reduce(
```

Comment:
Is there a use case in the app in which this function would be called with an empty exchanges or balances? If yes, it would be more performant to not iterate over the tokens in those cases.

Reply:
Just to be sure, I tested it, and we can indeed add a check to not go through tokens if some of the required data is not yet available.

Reply:
I applied this comment as well as added return type here - https://github.com/dfinity/oisy-wallet/pull/2264